### ISSUE TITLE
bug: `t.drop(t.a)` unexpectedly selects `BitwiseNot(a)` instead of selecting the other columns

### ISSUE THREAD

**Issue by ogrisel** [ibis-project/ibis#6872]
### What happened?

Reproducer setup:

```python
import ibis

t = ibis.memtable({"a": [1], "b": [0.1]})
t.drop("a")
```

works as expected (selects column "b" and discards column "a"):

```
r0 := InMemoryTable
  data:
    PandasDataFrameProxy:
         a    b
      0  1  0.1

Selection[r0]
  selections:
    b: r0.b
```

however, the following does not do the same, and attempts to perform an `BitwiseNot` operation instead:

```python
t.drop(t.a)
```

```
r0 := InMemoryTable
  data:
    PandasDataFrameProxy:
         a    b
      0  1  0.1

Selection[r0]
  selections:
    BitwiseNot(a): BitwiseNot(r0.a)
```

For column types for which the negation operator is not valid, this causes the following crash:

```python
t.drop(t.b)
```
```python-traceback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 1
----> 1 t.drop(t.b)

File ~/code/ibis/ibis/expr/types/relations.py:1940, in Table.drop(self, *fields)
   1937     raise KeyError(f"Fields not in table: {sorted(missing_fields)}")
   1939 sels = (s.c(f) if isinstance(f, str) else f for f in fields)
-> 1940 return self.select(~s.any_of(*sels))

TypeError: bad operand type for unary ~: 'FloatingColumn'
```

### What version of ibis are you using?

6.1.0 (actually the dev source code from today).

### What backend(s) are you using, if any?

DuckDB but it seems independent of the backend.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
Thanks @ogrisel!

Yeah, we should definitely allow column references in drop.

**Comment by gforsyth** [comment]
Fixed in #6879

**Comment by gforsyth** [comment]
Whoops, thought I forgot to link the issue in the PR, but it looks like CI is stalled -- reopening for the moment.
